crates/binstalk/src/helpers/git.rs

```diff
@@ -53,39 +89,87 @@ impl FromStr for GitUrl {
 }
 
 #[derive(Debug)]
-pub struct Repository(gix::Repository);
+pub struct Repository(gix::ThreadSafeRepository);
 
 impl Repository {
+    fn prepare_fetch(
+        url: GitUrl,
+        path: &Path,
+        kind: create::Kind,
+    ) -> Result<clone::PrepareFetch, GitError> {
+        Ok(clone::PrepareFetch::new(
+            url.0,
+            path,
+            kind,
```

Comment:
Now I see of `--git` is used and why it would be better to not use `bare` clones for these to assure support symlinks and all.

Reply:
It's not just symlink support, users could also provide a mono repository containing a giant workspace, with some crates inside submodule for whatever reason.

Personally I think we should support whatever repository `--git` passed, but for `--registry` we can assume its structure and it's likely there won't be any symlink or submodules.

Besides, I would have to rewrite `load_manifest_from_workspace` and in order to retain the existing test cases, I would have to implement a new `Vfs` trait.

If it won't work with submodules, then it probably worth my effort supporting it.

Reply:
BTW @Byron does `cargo` uses the same technique here to avoid checking out repository?

Reply:
> If it won't work with submodules, then it probably worth my effort supporting it.

Speaking of submodules, `gitoxide` also can't check them out right now, but that will be what I am working on next after getting through my backlog.

> [..] does `cargo` uses the same technique here to avoid checking out repository?

Yes, even though it still uses `git2` for that.